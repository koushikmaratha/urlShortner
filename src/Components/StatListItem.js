import React from 'react';

const StatListItem = (props) => {
    let { long_url, short_url, count, countries } = props.statData;
    return (
        <div>
            <li className="list-group-item">
                <p style={{fontWeight:'bold'}}>{long_url}</p>
                <p>{`${window.location.origin.toString()}/${short_url}`}</p>
                <span>
                    <p>Total Clicks: {count}</p>
                    <p>Top Counties: {countries}</p>
                </span>
            </li>
        </div>
    );
};

export default React.memo(StatListItem);