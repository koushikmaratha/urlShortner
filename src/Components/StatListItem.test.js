import React from 'react';
import { shallow } from 'enzyme';
import StatListItem from './StatListItem';


describe('Test the StatListItem Component', () => {
    let wrapper;
    beforeEach(() => (wrapper = shallow(<StatListItem statData={{}} />)));

    it('Test the Component with no Data', () => {
        expect(wrapper.find('p')[0]).toBe(undefined);
    })

    it('Tests the SnapShot',()=>{
        wrapper.setProps({statData:{long_url:'http://www.google.com',short_url:'kskKSK', count:20, countries:'India, Russia'}});
        expect(wrapper).toMatchSnapshot();
    })
})