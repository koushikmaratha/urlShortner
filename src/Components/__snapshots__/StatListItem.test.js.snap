// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the StatListItem Component Tests the SnapShot 1`] = `ShallowWrapper {}`;
